Lib/encodings/__init__.py

```diff
@@ -156,19 +156,50 @@ def search_function(encoding):
 codecs.register(search_function)
 
 if sys.platform == 'win32':
-    # bpo-671666, bpo-46668: If Python does not implement a codec for current
-    # Windows ANSI code page, use the "mbcs" codec instead:
-    # WideCharToMultiByte() and MultiByteToWideChar() functions with CP_ACP.
-    # Python does not support custom code pages.
-    def _alias_mbcs(encoding):
+    def _code_page_search_function(encoding):
+        encoding = encoding.lower()
+        if not encoding.startswith('cp'):
+            return None
         try:
-            import _winapi
-            ansi_code_page = "cp%s" % _winapi.GetACP()
-            if encoding == ansi_code_page:
-                import encodings.mbcs
-                return encodings.mbcs.getregentry()
-        except ImportError:
-            # Imports may fail while we are shutting down
-            pass
-
-    codecs.register(_alias_mbcs)
+            cp = int(encoding[2:])
```

Comment:
Can you please move the codec code to a new module, e.g. _win_cp_codecs.py ? Codec definitions don't belong into the __init__.py of the package. The search function can stay in this module.

Reply:
Note that this is not a single codec, but a family of codecs, parametrized by the code page number. So we should move a factory function that creates a codec for the code page. Or move the search function.

Reply:
Yes, I know, and the codecs are dynamically generated, which is fine as well, but this can still be put into a separate module. I'm not a friend of putting lots of code into `__init__.py` files. 

The search functions need to be here, since we have to make sure that they are registered when importing the encodings package, but codec code should live in separate modules.